### ISSUE TITLE
macOS error when adding a new additional header and then deleting it

### ISSUE THREAD

**Issue by grndis** [Julien-cpsn/ATAC#119]
![CleanShot 2024-10-27 at 21 32 39@2x](https://github.com/user-attachments/assets/15a7bbec-8879-4ce4-a454-bf4b7a59e1e5)

I got this error when adding a new header then delete that additional header. Deleting the default header is okay, but for the newly added header, I encountered an error that caused a force close.
```
 thread 'main' panicked at src/app/business_logic/request/utils.rs:15:25:
index out of bounds: the len is 1 but the index is 5                                                                                  │
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace             
```

**Comment by Julien-cpsn** [comment]
Fixed in https://github.com/Julien-cpsn/ATAC/commit/e3287f2bec54b8088df7bd55699bc4dc787c5eb9

**Comment by grndis** [comment]
> Fixed in [e3287f2](https://github.com/Julien-cpsn/ATAC/commit/e3287f2bec54b8088df7bd55699bc4dc787c5eb9)

@Julien-cpsn Thank you! The error has been resolved. However, after I delete a header, the selection reverts back to the first header. It would be helpful if the selection could instead update to the last header prior to deletion. This way, you wouldn't have to use repetitive keystrokes when removing multiple headers.
